rkt/fetch_test.go

```diff
@@ -152,19 +253,67 @@ func TestDownloading(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write(body)
-	}))
-	defer ts.Close()
-
+	noauthServer := &serverHandler{
+		body: body,
+		t:    t,
+		auth: "none",
+	}
+	basicServer := &serverHandler{
+		body: body,
+		t:    t,
+		auth: "basic",
+	}
+	oauthServer := &serverHandler{
+		body: body,
+		t:    t,
+		auth: "bearer",
+	}
+	denyServer := &serverHandler{
+		body: body,
+		t:    t,
+		auth: "deny",
+	}
+	noAuthTS := httptest.NewTLSServer(noauthServer)
+	defer noAuthTS.Close()
+	basicTS := httptest.NewTLSServer(basicServer)
+	defer basicTS.Close()
+	oauthTS := httptest.NewTLSServer(oauthServer)
+	defer oauthTS.Close()
+	denyAuthTS := httptest.NewServer(denyServer)
+	noAuth := http.Header{}
+	// YmFyOmJheg== is base64(bar:baz)
+	basicAuth := http.Header{"Authorization": {"Basic YmFyOmJheg=="}}
+	bearerAuth := http.Header{"Authorization": {"Bearer sometoken"}}
+	urlToName := map[string]string{
+		noAuthTS.URL:   "no auth",
+		basicTS.URL:    "basic",
+		oauthTS.URL:    "oauth",
+		denyAuthTS.URL: "deny auth",
+	}
 	tests := []struct {
-		ACIURL string
-		SigURL string
-		body   []byte
-		hit    bool
+		ACIURL   string
+		SigURL   string
```

Comment:
Yeah, I was thinking about it when I was working with this code. In the end I decided against it, because maybe somebody sometime will want to test fetching code with signatures. But that probably can be done by another test case.